bindings/experimental/distrdf/python/DistRDF/Ranges.py

```diff
@@ -13,16 +14,28 @@ class EmptySourceRange(object):
 
     Attributes:
 
+    id (int): Sequential counter to identify this range. It is used to assign
+        a filename to a partial Snapshot in case it was requested. The id
+        counter is reset to zero whenever a new RDataFrame is created in the
```

Comment:
Is this true? What happens if I call again Snapshot on an RDataFrame that already exists? It should have reset the counter, right?
If there were a counter per RDF object, we could just make the counter a (hidden) attribute of the RDF object, but I think this is not true. It has to be reset every time we call Snapshot.

Reply:
No in fact I miswrote the documentation. Since `reset_index` is called in `get_*_ranges`, it will get called whenever a new Snapshot is called. Even on the same RDF.

Reply:
That happens because the ranges are recomputed each time there's the need to run another event loop even with the same RDF